packages/opentelemetry-exporter-collector/src/CollectorExporterBase.ts

```diff
@@ -132,14 +143,106 @@ export abstract class CollectorExporterBase<
     return this._shuttingDownPromise;
   }
 
+  private _getUrl(config: T): string {
+    // if given, use url defined in config
+    if (typeof config.url !== 'undefined') {
+      return config.url;
+    }
+    const endpointFromEnv = getEnv().OTEL_EXPORTER_OTLP_ENDPOINT;
+    const defaultEndpoint = this.getProtocol().includes('http')
+      ? `http://${this._defaultEndpoint}/v1/${this.getExporterType()}s`
+      : this._defaultEndpoint;
+    console.log(this.getExporterType(), this.getProtocol(), defaultEndpoint)
+    switch (this.getExporterType()) {
+      case 'metric': {
+        const url = getEnv().OTEL_EXPORTER_OTLP_METRICS_ENDPOINT;
+        return url && url.length > 0
+          ? url
+          : endpointFromEnv && endpointFromEnv.length > 0
+          ? endpointFromEnv
+          : defaultEndpoint;
+      }
+      case 'trace': {
+        const url = getEnv().OTEL_EXPORTER_OTLP_TRACES_ENDPOINT;
+        return url && url.length > 0
+          ? url
+          : endpointFromEnv && endpointFromEnv.length > 0
+          ? endpointFromEnv
+          : defaultEndpoint;
+      }
+      default: {
+        return getEnv().OTEL_EXPORTER_OTLP_ENDPOINT;
+      }
+    }
+  }
+
+  private _getDefaultServiceName(config: T): string {
```

Comment:
keeping this in base will get extra overhead for each instance. Like for example metric exporter doesn't need to know the name of trace exporter and opposite. Plus each instance will have extra function for just getting it's name. Whereas you can keep the final name in the specific class.